### PR TITLE
fix(scan) reject ballots with the wrong precinct

### PR DESCRIPTION
**Pull request by carolinemodic**
## Overview

Implements enforcing the precinct that gets set when scanning for VxScan. 

@jonahkagan I didn't really understand the layout of the code here and what is for precinct-scanner or the general interpreter so I easily may have done this in the wrong spot. 

Will make a seperate PR for the merge to m14-release-branch as there are an annoying amount of conflicts. 
https://github.com/votingworks/vxsuite/issues/2438
## Testing Plan 
Added automated tests
Manually tests configuring for a precinct verifying that precinct ballots (bmd and hmpb) can scan and other precincts can not. Changed precinct to a different precinct, repeated. Changed back to all precincts and verified everything could scan. Also tested that killing and restarting the app didn't change how things worked. 

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Review by arsalansufi — APPROVED** [review]
Looks good to me at a high level! Approving to unblock, though we can still get feedback from others who have more context on this code

**Review by jonahkagan — APPROVED** [review]
Looks good to me! I'd defer to @eventualbuddha for reviewing the actual interpreter section.

Do we need to also implement this for the NH interpreter?

**Review by eventualbuddha — APPROVED** [review]
> Do we need to also implement this for the NH interpreter?

Yes, the changes in this PR do not apply to the NH code path. Make a ticket to follow up?

**Comment by carolinemodic** [comment]
Created https://github.com/votingworks/vxsuite/issues/2445 for the nh work

**Pull request merged**
